src/syncer.ts

```diff
@@ -0,0 +1,726 @@
+import fs from "fs";
+
+import { Octokit } from "@octokit/rest";
+import { createAppAuth } from "@octokit/auth-app";
+import winston from "winston";
+
+import { Database, PRAction, Repo } from "./db";
+import { applyPatches } from "./git";
+import { createFailureIssue } from "./github";
+
+const DEFAULT_CONFIG_PATH = "/etc/magic-mirror/config.json";
+const DEFAULT_PRIVATE_KEY_PATH = "/etc/magic-mirror/auth.key";
+
+/**
+ * Config is the deserialized form of the user provided configuration.
+ */
+export type Config = {
+  appID: number;
+  dbPath?: string;
+  logLevel?: string;
+  privateKeyPath?: string;
+  upstreamMappings: {
+    // Fork organization
+    [key: string]: {
+      // Upstream organization
+      [key: string]: {
+        branchMappings: {
+          // Upstream branch to Fork branch
+          [key: string]: string;
+        };
+      };
+    };
+  };
+};
+
+/**
+ * Syncer is the backend of Magic Mirror, which monitors upstream and creates PRs on forks.
```

Comment:
I can't help but think of this as "sinker" 😆

Reply:
I'm open to other names.  :smile: